agent/taskresource/grpcclient/credentialsfetcherclient_test.go

```diff
@@ -78,72 +99,180 @@ func TestCredentialsFetcherClient_AddKerberosLease(t *testing.T) {
 		name             string
 		credspecContents []string
 		response         CredentialsFetcherResponse
+		expectedError    string
+	}{
+		{
+			"invalid request empty credspec contents",
+			[]string{},
+			CredentialsFetcherResponse{},
+			"rpc error: code = InvalidArgument desc = credentialspecs should not be empty",
+		},
+		{
+			"valid request credspecs associated to gMSA account",
+			[]string{credspec_webapp01},
+			CredentialsFetcherResponse{LeaseID: leaseid, KerberosTicketPaths: []string{"/var/credentials-fetcher/krbdir/123456/webapp01", "/var/credentials-fetcher/krbdir/123456/webapp02"}},
+			"",
+		},
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.DialContext(ctx, "", grpc.WithInsecure(), grpc.WithContextDialer(dialer()))
+	require.NoError(t, err)
+	defer conn.Close()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			response, err := NewCredentialsFetcherClient(conn, time.Minute).AddKerberosLease(context.Background(), tt.credspecContents)
+			if tt.expectedError != "" {
+				assert.Error(t, err)
+				assert.Equal(t, tt.expectedError, err.Error())
```

Comment:
Code will panic at `err.Error()` if `err` is `nil`. How about merging these two asserts into a single `assert.EqualError(t, err, tt.expectedError)`? That will perform both the assertions safely. Another option is to use `require.Error(t, err)` for the first assertion so that the test stops if that assertion fails. 

Same for `TestCredentialsFetcherClient_AddNonDomainJoinedKerberosLease` and `TestCredentialsFetcherClient_RenewNonDomainJoinedKerberosLease`.